cmd/thanos/check.go

```diff
@@ -0,0 +1,113 @@
+package main
+
+import (
+	"fmt"
+	"io/ioutil"
+	"os"
+
+	"github.com/go-kit/kit/log"
+	thanosrule "github.com/improbable-eng/thanos/pkg/rule"
+	"github.com/oklog/run"
+	"github.com/opentracing/opentracing-go"
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/prometheus/prometheus/pkg/rulefmt"
+	"gopkg.in/alecthomas/kingpin.v2"
+	"gopkg.in/yaml.v2"
+)
+
+func registerChecks(m map[string]setupFunc, app *kingpin.Application, name string) {
+	cmd := app.Command(name, "Linting tools for Thanos")
+	registerRulesCheck(m, cmd, name)
+	return
+}
+
+func registerRulesCheck(m map[string]setupFunc, root *kingpin.CmdClause, name string) {
+	checkRulesCmd := root.Command("rules", "Check if the rule files are valid or not.")
+	ruleFiles := checkRulesCmd.Arg(
+		"rule-files",
+		"The rule files to check.",
+	).Required().ExistingFiles()
+
+	m[name+" rules"] = func(g *run.Group, logger log.Logger, reg *prometheus.Registry, _ opentracing.Tracer, _ bool) error {
+		// Dummy actor to immediately kill the group after the run function returns.
+		g.Add(func() error { return nil }, func(error) {})
+		return checkRulesFiles(ruleFiles)
+	}
+}
+
+func checkRulesFiles(files *[]string) error {
+	var failed error
+
+	for _, f := range *files {
+		if n, errs := checkRules(f); errs != nil {
+			fmt.Fprintln(os.Stderr, "  FAILED:")
+			for _, e := range errs {
+				fmt.Fprintln(os.Stderr, e.Error())
+			}
+			failed = fmt.Errorf("Errors: %v", errs)
+		} else {
+			fmt.Printf("  SUCCESS: %d rules found\n", n)
+		}
+		fmt.Println()
+	}
+	if failed != nil {
+		return failed
+	}
+	return nil
+}
+
+func checkRules(filename string) (int, []error) {
```

Comment:
worth to just return `error` and use  `tsdb.Multierrror`?

Reply:
Great, didn't know about the tsdb multierror. Will change thanks.

Reply:
done, switched to multierror (just a heads up in new version of TSDB multierror moves from generic tsdb package to tsdb/errors package. Imported if according to godoc and then gomod completely broke the dependencies since it upgraded tsdb to new version since the old did not had the package :D )